### PR TITLE
changes class name CustomPaymentEntry to CheckRunPaymentEntry

### PR DESCRIPTION
**Pull request by viralkansodiya**
Issue No #193

#hooks.py #payment_entry.py
Changes in class name of payment entry

#check_run.py
Paid invoices and expense claim should not be listed on check run(HTML Section)

#check_run.vue

before fix bug
on change event of mode of payment ** Process Check Run** Button  not convert in save button

after fixing
[29_1_2024, 10_07_15 am - Screen - Untitled video.webm](https://github.com/agritheory/check_run/assets/141210323/7ca0add4-3032-4231-8aee-c269cf9f6ce4)

### CODE REVIEW

**Comment by viralkansodiya** [comment]
@agritheory Please look at this Pr we have fixed some bug as we observed

File: check_run/check_run/doctype/check_run/check_run.py [hunk 1]
```diff
@@ -532,6 +532,7 @@ def get_entries(doc: CheckRun | str) -> dict:
 		.where(Coalesce(payment_schedule.outstanding, purchase_invoices.outstanding_amount) != 0)
 		.where(purchase_invoices.company == company)
 		.where(purchase_invoices.docstatus == 1)
+		.where(purchase_invoices.status != 'Paid')
```
Comment (agritheory) [inline check_run/check_run/doctype/check_run/check_run.py]: Using a string literal here makes this incompatible with Workflows

File: check_run/check_run/doctype/check_run/check_run.py [hunk 2]
```diff
@@ -564,6 +565,7 @@ def get_entries(doc: CheckRun | str) -> dict:
 		.where(exp_claims.grand_total > exp_claims.total_amount_reimbursed)
 		.where(exp_claims.company == company)
 		.where(exp_claims.docstatus == 1)
+		.where(exp_claims.status == 'Paid')
```
Comment (agritheory) [inline check_run/check_run/doctype/check_run/check_run.py]: Using a string literal here makes this incompatible with Workflows

File: check_run/public/js/check_run/CheckRun.vue [hunk 3]
```diff
@@ -216,6 +217,19 @@ function update_sort(key_name) {
 }
 
 function onMOPChange(event, rowName) {
+	// change of mode of payment make document Unsaved
+	//start
+	var element1 = document.querySelector(`#${rowName}`)
+	element1.checked = !element1.checked
+	console.log(element1.checked)
```
Comment (agritheory) [inline check_run/public/js/check_run/CheckRun.vue]: No console, please

File: check_run/public/js/check_run/CheckRun.vue [hunk 4]
```diff
@@ -216,6 +217,19 @@ function update_sort(key_name) {
 }
 
 function onMOPChange(event, rowName) {
+	// change of mode of payment make document Unsaved
+	//start
+	var element1 = document.querySelector(`#${rowName}`)
```
Comment (agritheory) [inline check_run/public/js/check_run/CheckRun.vue]: This is not a performant way to handle this, it adds too many listeners and crashes when we see > 1000 documents in the Check Run.

File: check_run/public/js/check_run/CheckRun.vue [hunk 5]
```diff
@@ -216,6 +217,19 @@ function update_sort(key_name) {
...
+	element1.checked = !element1.checked
+	console.log(element1.checked)
+	var changeEvent1 = new Event('change');
+	element1.dispatchEvent(changeEvent1);
+	var element2 = document.querySelector(`#${rowName}`)
+	element2.checked = !element1.checked
+	var changeEvent2 = new Event('change');
```
Comment (agritheory) [inline check_run/public/js/check_run/CheckRun.vue]: We want to be idiomatic about using Vue or using the Frappe form API, follwing the conventions set by Frappe (the comapny). This vanilla JS approach, while valid and normally considered to be the most universally readable approach, isn't either of those.

Comment (agritheory) [reply check_run/public/js/check_run/CheckRun.vue]: `frm.dirty()`

**Comment by viralkansodiya** [comment]
@agritheory In this PR now only Class name changes are available

**Comment by viralkansodiya** [comment]
@agritheory I am closing this pull request and will do another PR

**Pull request closed without merging**